jcommune-service/src/test/java/org/jtalks/jcommune/service/transactional/TransactionalUserContactsServiceTest.java

```diff
@@ -104,6 +104,26 @@ public void saveEditedContactsShouldUpdateEditedContacts() throws NotFoundExcept
     }
 
     @Test
+    public void saveEditedContactsWithNotEmptyValue() throws NotFoundException {
+        long contactTypeId1 = 1;
+        long contactTypeId2 = 2;
+        String newValue2 = null;
+
+        List<UserContactContainer> contacts = new ArrayList<>(addContactsToUser(user, 2, contactTypeId1));
+        UserContactContainer contact1 = contacts.get(0);
+        contact1.setTypeId(contactTypeId2);
+        contact1.setValue(newValue2);
+
+        prepareContactsMocks();
+        UserContactType contactType = createUserContactType(contactTypeId2);
+        when(userContactsDao.get(contactType.getId())).thenReturn(contactType);
+        when(userContactsDao.isExist(anyLong())).thenReturn(true);
+
+        JCUser resultUser = userContactsService.saveEditedUserContacts(user.getId(), contacts);
+        assertEquals(resultUser.getContacts().size(), 1,"Contact with not empty value haven't remove");
```

Comment:
If that's what you're testing, then the test name is actually wrong. The name says you're testing non-empty contacts while you're actually testing the empty ones..